File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
      int red = 0, white = 0, blue = 0;
      for (int i = 0; i < nums.length; i++) {
        red += nums[i] == 0 ? 1 : 0;
        white += nums[i] == 1 ? 1 : 0;
        blue += nums[i] == 2 ? 1 : 0;
      }  
      for (int i = 0; i < red; i++) {
        nums[i] = 0;
      }
      for (int i = 0; i < white; i++) {
        nums[red + i] = 1;
      }
      for (int i = 0; i < blue; i++) {
        nums[red + white + i] = 2;
      }
    }
}

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # l, idx, r = 0, 0, len(nums) - 1
        # while idx <= r:
        #     if nums[idx] == 0: 
        #         nums[idx], nums[l] =  nums[l], nums[idx]
        #         l += 1
        #         idx += 1
        #     elif nums[idx] == 2:
        #         nums[idx], nums[r] =  nums[r], nums[idx]
        #         r -= 1
        #     else:
        #         idx += 1            
        map = {}
        for num in nums:
            map[num] = map.get(num, 0) + 1

        idx = 0
        for clor in range(3):
            for _ in range(map.get(clor, 0)):
                nums[idx] = clor
                idx += 1

        